flink-runtime/src/main/java/org/apache/flink/runtime/blob/BlobServerSocket.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.blob;
+
+import org.apache.flink.configuration.BlobServerOptions;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.configuration.JobManagerOptions;
+import org.apache.flink.configuration.SecurityOptions;
+import org.apache.flink.runtime.net.SSLUtils;
+import org.apache.flink.util.NetUtils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.net.ServerSocketFactory;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.ServerSocket;
+import java.util.Collections;
+import java.util.Iterator;
+
+public class BlobServerSocket {
+
+    private static final Logger LOG = LoggerFactory.getLogger(BlobServerSocket.class);
+
+    private final Configuration config;
+    private final int backlog;
+    private final String serverPortRange;
+    private ServerSocket serverSocket;
+    private final int maxConnections;
+    private boolean firstCreation;
+
+    public BlobServerSocket(Configuration config, int backlog, int maxConnections)
+            throws IOException {
+        this.config = config;
+        this.backlog = backlog;
+        this.maxConnections = maxConnections;
+
+        serverPortRange = config.get(BlobServerOptions.PORT);
+        this.firstCreation = true;
+        createSocket();
+        this.firstCreation = false;
+    }
+
+    public ServerSocket getServerSocket() {
+        return serverSocket;
+    }
+
+    public void createSocket() throws IOException {
+        Iterator<Integer> ports;
+        if (firstCreation) {
```

Comment:
create this field with initialized true and set firstCreation to the false once socket is created.

Reply:
not sure about what you expect but I've changed just let me know if this is fine